### PR TITLE
Fix reclaimUnallocatedRewards() can't be executed unless every user unstakes

### PR DESCRIPTION
**Pull request by alexanderattar**
# Fix reclaimUnallocatedRewards Being Blocked by Active Stakers (Issue #5)

## Problem

When `reclaimUnallocatedRewards()` was called with active stakers remaining, it would revert even if those stakers had emergency exited and forfeited their pending rewards. This prevented the owner from reclaiming legitimately forfeited rewards in normal operational scenarios.

## Solution

Implemented the exact fix recommended by the audit: track forfeited rewards separately and enhance the existing `reclaimUnallocatedRewards()` function to prioritize forfeited rewards. Now the function can reclaim forfeited rewards even with active stakers, while preserving original protections.

### Contract Changes (Following Audit Recommendation)
- **Added forfeitedRewards state variable**: Tracks rewards forfeited via emergency exit
- **Modified emergencyExit()**: Now tracks forfeited pending rewards with proper accounting
- **Enhanced reclaimUnallocatedRewards()**: Prioritizes forfeited rewards, maintains original behavior for unallocated rewards

### Test Updates
- **5 new tests**: Cover forfeited rewards tracking, reclaim functionality, and edge cases
- **All 81 tests passing**: No regressions in existing functionality

## Impact

- **Audit Compliance**: Implements exact fix recommended by security audit
- **Operational Flexibility**: Owner can now reclaim forfeited rewards with active stakers remaining
- **Backward Compatibility**: Enhanced existing function maintains all previous functionality

### CODE REVIEW

File: src/StakingRewards.sol [hunk 1]
```diff
@@ -501,23 +512,44 @@ contract StakingRewards is Ownable2Step, ReentrancyGuard, Pausable {
...
+            reclaimAmount = forfeited;
+            forfeitedRewards = 0;
+        } else if (activeStake == 0) {
+            // Can only reclaim full unallocated when no active stakers
+            uint256 unallocated = unallocatedRewards;
+            if (unallocated == 0) revert ZeroAmount();
+            reclaimAmount = unallocated;
+            unallocatedRewards = 0;
+        } else {
+            revert ActiveStakeExists();
```
[the remark this answers is not in the document]
Comment (alexanderattar) [reply src/StakingRewards.sol:537]: we do already have one for this here https://github.com/holographxyz/holograph-protocol/blob/5ac3be26f16b5bf3c705e953d85ac718b8d2dc54/test/StakingRewards.t.sol#L650C14-L650C65
